build/helper/parameter_usage_options.py

```diff
@@ -5,32 +5,36 @@ class ParameterUsageOptions(Enum):
     '''Different usage options for parameter lists.'''
 
     SESSION_METHOD_DECLARATION = 1
```

Comment:
We should [just use auto](https://docs.python.org/3/library/enum.html#using-automatic-values) here.

Reply:
`auto()`  is only available in 3.6 and the enum backport. I used the solution given in the 3.5 documentation:
```python
class AutoNumber(Enum):
    def __new__(cls):
        value = len(cls.__members__) + 1
        obj = object.__new__(cls)
        obj._value_ = value
        return obj


class ParameterUsageOptions(AutoNumber):
    '''Different usage options for parameter lists.'''

    SESSION_METHOD_DECLARATION = ()
    '''For declaring a regular method in Session'''
    SESSION_INIT_DECLARATION = ()```